cmd/erasure-healing-common.go

```diff
@@ -26,29 +26,39 @@ import (
 )
 
 // commonTime returns a maximally occurring time from a list of time.
-func commonTime(modTimes []time.Time) (modTime time.Time) {
-	timeOccurenceMap := make(map[int64]int, len(modTimes))
+func commonTimeAndOccurence(times []time.Time, resolution time.Duration) (maxTime time.Time, maxima int) {
+	timeOccurenceMap := make(map[int64]int, len(times))
 	// Ignore the uuid sentinel and count the rest.
-	for _, t := range modTimes {
+	for _, t := range times {
 		if t.Equal(timeSentinel) {
 			continue
 		}
-		timeOccurenceMap[t.UnixNano()]++
+		nano := t.UnixNano()
+		if resolution > 0 {
+			nano = t.Round(resolution).UnixNano()
```

Comment:
Again, you cannot use rounding, since entries milliseconds apart will be in different "buckets"

For example "(2.49).Round(5)" -> 0 "(2.51).Round(5)" -> 5.